lib/ansible/module_utils/urls.py

```diff
@@ -686,7 +685,7 @@ def http_request(self, req):
                 else:
                     raise ProxyError('Unsupported proxy scheme: %s. Currently ansible only supports HTTP proxies.' % proxy_parts.get('scheme'))
             else:
-                s.connect((self.hostname, self.port))
+                s = socket.create_connection((self.hostname, self.port))
```

Comment:
@jimi-c made the following suggestion:

Since we're going to be stopping support of python-2.4 in new versions of Ansible in April of next year, perhaps it makes sense to use create_connection() if  it's available, otherwise using socket.connect().  Would also need a note that the code won't work with ipv6 on python-2.4.